pkg/ccl/changefeedccl/encoder.go

```diff
@@ -321,8 +322,11 @@ type confluentAvroEncoder struct {
 	updatedField, beforeField, keyOnly bool
 	targets                            jobspb.ChangefeedTargets
 
-	keyCache      map[tableIDAndVersion]confluentRegisteredKeySchema
-	valueCache    map[tableIDAndVersionPair]confluentRegisteredEnvelopeSchema
+	keyCache   *cache.UnorderedCache // [tableIDAndVersion]confluentRegisteredKeySchema
+	valueCache *cache.UnorderedCache // [tableIDAndVersionPair]confluentRegisteredEnvelopeSchema
```

Comment:
question: tableID and version only changes when schema change happens.. And, we have invariant that only 2 versions of a descriptor can be live at any time.
Does it make sense to just delete everything when version < 2 previous ones?

Reply:
I don't think we have that invariant in this context. Only two versions can be live in respect to current versions, but a rangefeed could still emit an old version of row 1 after emitting a new version of row 2, with the number of schema changes in between being arbitrarily high.